hfinger/hfinger_core.py

```diff
@@ -197,26 +201,29 @@ def get_accept_language_value(hdr):
 def get_pop_hdr_val(pkt):
     r = []
     for i in pkt[1:]:
-        t = i.split(":")[0].lower()
-        if t == "connection":
-            r.append(get_hdr_value(i, "connection", CONNVAL))
-        elif t == "accept-encoding":
-            r.append(get_hdr_value(i, "accept-encoding", AEVAL))
-        elif t == "content-encoding":
-            r.append(get_hdr_value(i, "content-encoding", CONTENC))
-        elif t == "cache-control":
-            r.append(get_hdr_value(i, "cache-control", CACHECONT))
-        elif t == "te":
-            r.append(get_hdr_value(i, "te", TE))
-        elif t == "accept-charset":
-            r.append(get_hdr_value(i, "accept-charset", ACCPTCHAR))
-        elif t == "content-type":
-            r.append(get_content_type(i))
-        elif t == "accept":
-            r.append(get_hdr_value(i, "accept", ACCPT))
-        elif t == "accept-language":
-            r.append(get_accept_language_value(i))
-        elif t == "user-agent":
-            r.append(get_ua_value(i))
+        if ":" in i:
+            t = i.split(":")[0].lower()
+            if t == "connection":
+                r.append(get_hdr_value(i, "connection", CONNVAL))
+            elif t == "accept-encoding":
+                r.append(get_hdr_value(i, "accept-encoding", AEVAL))
+            elif t == "content-encoding":
+                r.append(get_hdr_value(i, "content-encoding", CONTENC))
+            elif t == "cache-control":
+                r.append(get_hdr_value(i, "cache-control", CACHECONT))
+            elif t == "te":
+                r.append(get_hdr_value(i, "te", TE))
+            elif t == "accept-charset":
+                r.append(get_hdr_value(i, "accept-charset", ACCPTCHAR))
+            elif t == "content-type":
+                r.append(get_content_type(i))
+            elif t == "accept":
+                r.append(get_hdr_value(i, "accept", ACCPT))
+            elif t == "accept-language":
+                r.append(get_accept_language_value(i))
+            elif t == "user-agent":
+                r.append(get_ua_value(i))
+        else:
+            logger.info("No colon in line: " + i)
```

Comment:
I think this will crash, trying to concatenate integer to a string.

Reply:
The variable naming was poor on my side - sorry. I've refactored the code and now it should be more clear that `i` (now `reqline`) is in fact a string. Check 048d938affe220984ed42d6ad371774af1d711c1